src/editor/TextEditor.java

```diff
@@ -1,11 +1,252 @@
 package editor;
 
 import javax.swing.*;
+import java.awt.*;
+import java.awt.event.WindowEvent;
+import java.awt.event.WindowListener;
 
 public class TextEditor extends JFrame {
+    static JTextArea textArea = new JTextArea();
+    static JTextField textField = new JTextField();
+    static String fileName;
+    static JCheckBox useRegex = new JCheckBox("Use regex");
+
+    private Color backColor = new Color(241, 241, 241);
+    private Font font = new Font(null,Font.BOLD,15);
+
+   // private Thread searchThread = new TextSearch();
+    TextSearch textSearch = new TextSearch();
+
     public TextEditor() {
+        Dimension dimension = new Dimension(600,500);
+
+        JMenuBar menu = menuCreator();
+        JPanel fileSelectGroup = createFileSelectGroup();
+        JScrollPane scrollPane = new JScrollPane(textArea);
+        addWindowListener(windowListener);
+        createLayout(menu, fileSelectGroup, scrollPane);
+
+        setSize(dimension);
+        setMinimumSize(dimension);
+        setTitle("Text editor v 0.5");
         setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
-        setSize(300, 300);
+        setLocationRelativeTo(null);
         setVisible(true);
     }
+
+    private JMenuBar menuCreator(){
+        JMenuBar menuBar = new JMenuBar();
+        JMenu file = menuFileCreator();
+        JMenu search = menuSearchCreator();
+        
+        file.setFont(font);
+        search.setFont(font);
+
+        menuBar.add(file);
+        menuBar.add(search);
+
+        return menuBar;
+    }
+
+    private JMenu menuFileCreator(){
+        JMenu file = new JMenu("File");
+
+        JMenuItem load = new JMenuItem("Open");
+        JMenuItem save = new JMenuItem("Save");
+        JMenuItem saveAs = new JMenuItem("Save as");
+        JMenuItem exit = new JMenuItem("Exit");
+
+        load.addActionListener( actionEvent -> FileChooser.open() );
+
+        save.addActionListener( actionEvent ->
+                FileLoader.saveFile( fileName, textArea.getText(), true));
+
+        saveAs.addActionListener( actionEvent ->
+                FileLoader.saveFile( null, textArea.getText(), true));
+
+        exit.addActionListener( actionEvent ->
+                {
+                    FileLoader.saveFile( fileName, textArea.getText(), true);
+                    dispose();
```

Comment:
Do you solve your problem with dispose?

Reply:
I've simplified code enough in future commits, so I didn't really need to call dispose() from outside .
But if for example the TextEditor is basic class of UI that contains window and an instance of MenuBar class(which contains exit button), I don't know how to call "dispose" method of TextEditor class to close the window of TextEditor. I've tried to make TextEditor static and create closeWindow method with dispose() inside, but it not calling from static context. Obviously exists more simple ways to solve this, but I've didn't see them yet.